src/mme-app/mmeStates/createBearerWfDedActComplete.cpp

```diff
@@ -0,0 +1,77 @@
+
+/*
+ * Copyright 2020-present Infosys Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+/**************************************
+ * createBearerWfDedActComplete.cpp
+ * This is an auto generated file.
+ * Please do not edit this file.
+ * All edits to be made through template source file
+ * <TOP-DIR/scripts/SMCodeGen/templates/stateMachineTmpls/state.cpp.tt>
+ **************************************/
+
+
+#include "actionTable.h"
+#include "actionHandlers/actionHandlers.h"
+#include "mmeSmDefs.h"
+#include "utils/mmeStatesUtils.h"
+#include "utils/mmeTimerTypes.h"
+
+#include "mmeStates/createBearerWfDedActComplete.h"
+
+using namespace mme;
+using namespace SM;
+
+/******************************************************************************
+* Constructor
+******************************************************************************/
+CreateBearerWfDedActComplete::CreateBearerWfDedActComplete():State(create_bearer_wf_ded_act_complete, 12000)
+{
+        stateEntryAction = &MmeStatesUtils::on_state_entry;
+        stateExitAction = &MmeStatesUtils::on_state_exit;
+        eventValidator = &MmeStatesUtils::validate_event;
+}
+
+/******************************************************************************
+* Destructor
+******************************************************************************/
+CreateBearerWfDedActComplete::~CreateBearerWfDedActComplete()
+{
+}
+
+/******************************************************************************
+* creates and returns static instance
+******************************************************************************/
+CreateBearerWfDedActComplete* CreateBearerWfDedActComplete::Instance()
+{
+        static CreateBearerWfDedActComplete state;
+        return &state;
+}
+
+/******************************************************************************
+* initializes eventToActionsMap
+******************************************************************************/
+void CreateBearerWfDedActComplete::initialize()
+{
+        {
+                ActionTable actionTable;
+                actionTable.addAction(&ActionHandlers::handle_ded_act_cmp_ind);
+                actionTable.addAction(&ActionHandlers::send_create_bearer_response);
+                actionTable.addAction(&ActionHandlers::create_bearer_proc_complete);
+                eventToActionsMap.insert(pair<uint16_t, ActionTable>(DED_ACT_COMPLETE, actionTable));
+        }
+        {
+                ActionTable actionTable;
+                actionTable.addAction(&ActionHandlers::handle_state_guard_timeouts);
+                eventToActionsMap.insert(pair<uint16_t, ActionTable>(STATE_GUARD_TIMEOUT, actionTable));
```

Comment:
On Guardtimeout, ABORT_EVENT is generated ? Because I dont see any code in handle_state_guard_timeouts function

Reply:
Taken care. We are now setting the MmeErrorCause to respond appropriately.